### ISSUE TITLE
implement pad/padding function

### ISSUE THREAD

**Issue by coquelin77** [helmholtz-analytics/heat#361]
**Feature functionality**
pad a function with constant values, other methods of padding are also useful

**Additional context**
https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.pad
one possible way to do this would be to simply pad the tensor with torch if the padding is in the non-split dimension, for the split dimension padding one could pad the begging/end then balance the tensor
